code/modules/reagents/machinery/chem_dispenser.dm

```diff
@@ -132,8 +135,8 @@
 /obj/machinery/chem_dispenser/ui_data(mob/user)
 	. = list()
 	.["amount"] = amount
-	.["energy"] = cell.charge ? cell.charge * powerefficiency : "0" //To prevent NaN in the UI.
-	.["maxEnergy"] = cell.maxcharge * powerefficiency
+	.["energy"] = cell ? cell.charge * powerefficiency : "0" //To prevent NaN in the UI.
+	.["maxEnergy"] = cell ? cell.maxcharge * powerefficiency : "0"
```

Comment:
Why is this not just a ?.

You also have this being sent as a string and float

Reply:
It needs to be 0 specifically for display stuff and ?. would give back null instead, didn't look too deeply into the ui it creates so I'm not certain how it's handling type.